### PR TITLE
[Enhancement] Enable transparent union rewrite by default

### PR DESCRIPTION
**Pull request by LiShuMing**
## Why I'm doing:

Since we have supportted transparent mv(https://github.com/StarRocks/starrocks/pull/42541 and https://github.com/StarRocks/starrocks/pull/43304), we can treat a mv as always-consistent, and no need consider its freshness  since the mv self will union refreshed data and no-refresh data.

For mv union rewrite, we can also no need consider mv's refreshness either.

eg: `test_mv1` can be used for mv rewrite whenever it's updated or not.

```
DROP MATERIALIZED VIEW test_mv1;
CREATE MATERIALIZED VIEW test_mv1 
PARTITION BY dt 
REFRESH DEFERRED MANUAL 
AS SELECT dt, sum(num) as num FROM t1 where dt > '2020-07-01' GROUP BY dt;

function: check_hit_materialized_view("SELECT dt, sum(num) as num FROM t1 where dt > '2020-07-01'  GROUP BY dt order by 1, 2 limit 3;", "test_mv1")
function: check_hit_materialized_view("SELECT dt, sum(num) as num FROM t1 where dt >'2020-07-01' GROUP BY dt having sum(num) > 10 order by 1, 2 limit 3;", "test_mv1")
function: check_hit_materialized_view("SELECT dt, sum(num) as num FROM t1 where dt > '2020-06-20'  GROUP BY dt order by 1, 2 limit 3;", "test_mv1", "UNION")
function: check_hit_materialized_view("SELECT dt, sum(num) as num FROM t1 where dt >'2020-06-20' GROUP BY dt having sum(num) > 10 order by 1, 2 limit 3;", "test_mv1", "UNION")
function: check_hit_materialized_view("SELECT dt, sum(num) as num FROM t1 GROUP BY dt order by 1, 2 limit 3;", "test_mv1", "UNION")
function: check_hit_materialized_view("SELECT dt, sum(num) as num FROM t1 GROUP BY dt having sum(num) > 10 order by 1, 2 limit 3;", "test_mv1", "UNION")
function: check_hit_materialized_view("SELECT * FROM (SELECT dt, sum(num) as num FROM t1 GROUP BY dt having sum(num) > 10 UNION ALL SELECT dt, sum(num) as num FROM t1 GROUP BY dt) t order by 1, 2 limit 3;", "test_mv1", "UNION")
```

## What I'm doing:
- Enable transparent (union) rewrite by default since we consider mv as always-consistent.
- Move `TRANSPARENT ` from UnionRewriteMode to SessionVariable since transaprent rewrite can be used associated with other union rewrite mode.
- Fix `explain costs` bugs if final tree contains `PhysicalProject`.

## Further
- Fix duplication bug in aggregate rewrite;
- Enable PULL_PREDICATE_V1/PULL_PREDICATE_V2 by default, since we can use mv as more scenes.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by satanson — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MaterializationContext.java [hunk 1]
```diff
@@ -186,6 +186,11 @@ public long getMVUsedCount() {
         return mvUsedCount;
     }
 
+    // whether this mv has been used multi times
+    public boolean isMvDuplicateUsed() {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MaterializationContext.java:190]:
```suggestion
    public boolean isMvReused() {
```

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java [hunk 2]
```diff
@@ -142,6 +142,9 @@ protected OptExpression viewBasedRewrite(RewriteContext rewriteContext, OptExpre
         EquationRewriter queryExprToMvExprRewriter =
                 buildEquationRewriter(mvProjection, rewriteContext, false);
 
+        // TODO:duplicate if mv has already outputted.
+        // mvOptExpr = duplicateMvOptExpression(rewriteContext, mvOptExpr, queryExprToMvExprRewriter);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java:146]: is it useless?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/OptExpressionDuplicator.java [hunk 3]
```diff
@@ -175,15 +183,31 @@ public OptExpression visitLogicalTableScan(OptExpression optExpression, Void con
...
+                        newPrunedPartitionPredicates.add(newPredicate);
+                    }
+                    olapScanBuilder.setPrunedPartitionPredicates(newPrunedPartitionPredicates);
+                }
+            } else {
+                try {
+                    ScanOperatorPredicates scanOperatorPredicates = scanOperator.getScanOperatorPredicates();
+                    scanOperatorPredicates.duplicate(rewriter);
+                } catch (AnalysisException e) {
+                    // ignore exception
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/OptExpressionDuplicator.java:209]: add a debug log?

File: fe/fe-core/src/main/java/com/starrocks/sql/Explain.java [hunk 4]
```diff
@@ -179,6 +179,11 @@ List<OperatorStr> buildChildOperatorStr(OptExpression optExpression, int step) {
             return childString;
         }
 
+        @Override
+        public OperatorStr visitPhysicalProject(OptExpression optExpression, OperatorPrinter.ExplainContext context) {
```
[the remark this answers is not in the document]
Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/sql/Explain.java:183]: Without this, `explain costs` may stackoverflow. just like `LogicalPlanPrinter`, add this.

```
        @Override
        public OperatorStr visit(OptExpression optExpression, OperatorPrinter.ExplainContext context) {
            return optExpression.getOp().accept(this, optExpression, context);
        }
```

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java (hunk 2 above)
Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java:146]: No. I will fix this later.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/OptExpressionDuplicator.java (hunk 3 above)
Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/OptExpressionDuplicator.java:209]: Fine, I will add it later.

**Review by ABingHuang — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Pull request merged**